ios/FluentUI/AvatarGroup/AvatarGroupTokens.swift

```diff
@@ -6,87 +6,95 @@
-        didSet {
-            if oldValue != size {
-                updateForCurrentTheme()
-            }
-        }
-    }
-
-    init(style: MSFAvatarGroupStyle,
-         size: MSFAvatarSize) {
+/// Design token set for the `AvatarGroup` control
+public class AvatarGroupTokens: ControlTokens {
+    /// Creates an instance of `AvatarGroupTokens`with optional token value overrides.
+    /// - Parameters:
+    ///   - interspace: CGFloat that defines the space between  the `Avatar` controls hosted by the `AvatarGroup`
+    ///   - ringInnerGap: CGFloat that defines the thickness of the space between the `Avatar` and its ring
+    ///   - ringThickness: CGFloat that defines the thickness of the ring around the `Avatar`
+    ///   - ringOuterGap: CGFloat that defines the thickness of the space around the ring of the `Avatar`
+    public init (style: MSFAvatarGroupStyle,
+                 size: MSFAvatarSize,
+                 interspace: CGFloat? = nil,
+                 ringInnerGap: CGFloat? = nil,
+                 ringThickness: CGFloat? = nil,
+                 ringOuterGap: CGFloat? = nil) {
         self.style = style
         self.size = size
 
         super.init()
 
-        self.themeAware = true
+        // Optional overrides
+        if let interspace = interspace {
+            self.interspace = interspace
+        }
 
-        updateForCurrentTheme()
-    }
+        if let ringInnerGap = ringInnerGap {
+            self.ringInnerGap = ringInnerGap
+        }
+
+        if let ringThickness = ringThickness {
+            self.ringThickness = ringThickness
+        }
 
-    @objc open func didChangeAppearanceProxy() {
-        updateForCurrentTheme()
+        if let ringOuterGap = ringOuterGap {
+            self.ringOuterGap = ringOuterGap
+        }
     }
 
-    override func updateForCurrentTheme() {
-        let currentTheme = theme
-        var appearanceProxy: AppearanceProxyType
-        let avatarTokens = currentTheme.MSFAvatarTokens
+    // MARK: - Design Tokens
+
+    let style: MSFAvatarGroupStyle
+    let size: MSFAvatarSize
 
+    lazy var interspace: CGFloat = {
         switch style {
         case .stack:
-            appearanceProxy = currentTheme.MSFAvatarGroupStackTokens
+            switch size {
+            case .xsmall, .small:
+                return -2 // -spacing[.xxxsmall]
+            case .medium:
+                return -4 // -spacing[.xxsmall]
+            case .large:
+                return -8 // -spacing[.xsmall]
+            case .xlarge, .xxlarge:
+                return -12 // -spacing[.small]
+            }
         case .pile:
-            appearanceProxy = currentTheme.MSFAvatarGroupTokens
+            switch size {
+            case .xsmall, .small:
+                return globalTokens.spacing[.xxSmall]
+            case .medium, .large, .xlarge, .xxlarge:
+                return globalTokens.spacing[.xSmall]
+            }
         }
+    }()
 
+    lazy var ringInnerGap: CGFloat = {
```

Comment:
This is a good point, and an indication that something is wrong in how we share tokens between `Avatar` and `AvatarGroup`. Let's discuss more offline with @sophialee0416 and @rdeassis, because I don't know that a cross-token relationship is appropriate here.